testcases/kernel/syscalls/getrandom/getrandom04.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2000 Silicon Graphics, Inc.  All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it would be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * Further, this software is distributed without any warranty that it is
+ * free of the rightful claim of any third person regarding infringement
+ * or the like.  Any license provided herein, whether implied or
+ * otherwise, applies only to this software file.  Patent licenses, if
+ * any, provided herein do not apply to combinations of this program with
+ * other software, or any other product whatsoever.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Contact information: Silicon Graphics, Inc., 1600 Amphitheatre Pkwy,
+ * Mountain View, CA  94043, or:
+ *
+ * http://www.sgi.com
+ *
+ * For further information regarding this notice, see:
+ *
+ * http://oss.sgi.com/projects/GenInfo/NoticeExplan/
+ */
+
+/*
+ * AUTHOR   : Cédric Hnyda
+ * DATE STARTED : 06/13/2015
+ *
+ *  Calls getrandom(2) after having limited the number of available file
+ *  descriptor to 3 and expects success.
+ */
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/fcntl.h>
+#include <sys/syscall.h>
+#include <errno.h>
+#include <string.h>
+#include <signal.h>
+#include <linux/random.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "test.h"
+#include "safe_macros.h"
+
+
+static void setup(void);
+static void cleanup(void);
+
+char *TCID = "getrandom04";
+int TST_TOTAL = 1;
+char *buf;
+int r;
+
+int main(int ac, char **av)
+{
+	tst_parse_opts(ac, av, NULL, NULL);
+
+	setup();
+	if ((r = system("ulimit -n 3") == -1))
+		tst_brkm(TBROK | TERRNO, cleanup, "ulimit -n 3 failed");
+	else {
+
```

Comment:
There is no need for the else branch, the ```tst_brkm()``` calls ```exit()``` and thus exits the program execution.